libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -790,10 +808,50 @@ bool ContractCompiler::visit(InlineAssembly const& _inlineAssembly)
 			_assembly.appendInstruction(Instruction::POP);
 		}
 	};
-	solAssert(_inlineAssembly.annotation().analysisInfo, "");
+
+	yul::Block const* code = &_inlineAssembly.operations();
+	yul::AsmAnalysisInfo analysisInfo = *_inlineAssembly.annotation().analysisInfo;
```

Comment:
```suggestion
	yul::AsmAnalysisInfo* analysisInfo = _inlineAssembly.annotation().analysisInfo.get()
```

then you do not need to copy and can use `*analysisInfo` below and replace `sharedCode` by the `Object` to hold the newly created analysis info in the "run optimizer" execution path.

Reply:
I had it like that in the previous iteration and thought having only a `yul::Block` outside the `if` instead of a `yul::Object` is slightly more lightweight. Changed it back now.